Supply live tracer

ziogaschr opened the pull request.

ziogaschr committed:
rename constant in comment

ziogaschr committed:
add BalanceChangeReason.String() for logging on development

ziogaschr committed:
add supply live tracer (draft)

ziogaschr committed:
simplify output

s1na commented on eth/tracers/live/supply.go:
```diff
@@ -0,0 +1,198 @@
+	diff := new(big.Int).Sub(newBalance, prevBalance)
+
+	switch reason {
+	case state.BalanceIncreaseGenesisBalance:
```
Here we're double-counting the genesis balances. Once here once in `OnGenesisBlock`

ziogaschr replied:
Great catch. After some point I was relying on the test and forgot that `BalanceIncreaseGenesisBalance` is not actually being called. To be on the safe side I removed the handling in **OnBalanceChange** and added a note.  [50ee919](https://github.com/s1na/go-ethereum/pull/21/commits/50ee9197bb9e0d0dbb9c8776987c73b84617a4d5)

ziogaschr committed:
TODO comment

ziogaschr committed:
Add test for supply live tracer

ziogaschr committed:
Merge branch 'extended-tracer' into extended-tracer-chris

ziogaschr committed:
/s/BalanceChangeWithdrawal/BalanceIncreaseWithdrawal/

ziogaschr committed:
fix Supply does not implement core.BlockchainLogger

ziogaschr committed:
Refactor supply live tracer tests

ziogaschr committed:
add test for selfdestruct

ziogaschr committed:
add supply tracer test where selfdestruct passes funds to itself

maoueh commented on core/state/metadata.go:
```diff
@@ -64,3 +64,42 @@ const (
+
+// TODO: to be removed, it's useful for logs while developing
+// String returns a string representation of the reason.
+func (r BalanceChangeReason) String() string {
```
Feels like this would be a good addition to the core code directly, it's probably worth it.

ziogaschr committed:
Count burned amount for selfdestructing contracts

ziogaschr commented on eth/tracers/internal/tracetest/supply_test.go:
```diff
@@ -0,0 +1,695 @@
+//   - Contract C selfdestructs and sends the funds to itself.
+//   - Contract D calls C and reverts (Burn amount of C
+//     has to be reverted as well).
+func TestSupplySelfdestructItselfAndRevert(t *testing.T) {
```
@s1na this tests handles both cases of counting burned amount when selfdestruct occurs + when it reverts.
For this reason I will remove the previous test.

ziogaschr replied:
Remove the referenced test with [694c51d](https://github.com/s1na/go-ethereum/pull/21/commits/694c51de943d7cf3c1a8927496a06da765b95daa)

ziogaschr commented on eth/tracers/live/supply.go:
```diff
@@ -0,0 +1,266 @@
+		calls: make([]supplyTxCallstack, 0),
+	}
+
+	// TODO shall we check if London enabled or b.BaseFee() != nil
```
@s1na I am thinking that I have to check if BaseFee is set and otherwise don't count the calue as Burned. Is this the case?

s1na replied:
No this "burn" is not the same as EIP-1559 burn. Here it's ether thats lost due to mechanics of selfdestruct which existed before.

ziogaschr commented on eth/tracers/live/supply.go:
```diff
@@ -0,0 +1,266 @@
+
+type Supply struct {
+	delta       SupplyInfo
+	txCallstack []supplyTxCallstack // Callstack for current transaction
```
@s1na for handling the **selfdestruct** to itself case, I used a similar logic to what we have at callTracer for internal transactions. This way, later we can handle more stuff on inner transactions if needed. What do you think?

s1na replied:
Yes that's how I would have done it as well.

ziogaschr committed:
remove obsolete test

s1na commented on eth/tracers/internal/tracetest/supply_test.go:
```diff
@@ -0,0 +1,560 @@
+			Alloc: core.GenesisAlloc{
+				addr1: {Balance: eth1},
+				aa: {
+					Code:    common.FromHex("0x73222222222222222222222222222222222222222273444444444444444444444444444444444444444460006000600060006000865af160006000600060006000865af150505050"),
```
Can you please share the source code also in a comment above the code?

ziogaschr replied:
Resolved by [ecca9df](https://github.com/s1na/go-ethereum/pull/21/commits/ecca9df80d461f3946f281b9ad1c82bcf4119c49)

s1na commented on eth/tracers/internal/tracetest/supply_test.go:
```diff
@@ -0,0 +1,560 @@
+					Balance: eth1,
+				},
+				dd: {
+					Code:    common.FromHex("0x73333333333333333333333333333333333333333360006000600060006000855af160006000fd5050"),
```
Same here

ziogaschr replied:
Resolved by [ecca9df](https://github.com/s1na/go-ethereum/pull/21/commits/ecca9df80d461f3946f281b9ad1c82bcf4119c49)

ziogaschr committed:
s/funds/eth1

ziogaschr committed:
remove TODO comment

ziogaschr committed:
stop handling BalanceChangeReason.BalanceIncreaseGenesisBalance

it is being handled in OnGenesisBlock

ziogaschr committed:
provide contracts code for TestSupplySelfdestructItselfAndRevert

ziogaschr committed:
add “—vmtrace.output.path” for setting the output path for traces

make use of lumberjack for rotating the trace logs every some MBs

ziogaschr committed:
fix misspell

ziogaschr committed:
remove debug logs

ziogaschr commented on eth/tracers/live/supply.go:
```diff
@@ -0,0 +1,245 @@
+	// Store traces in a rotating file
+	loggerOutput := &lumberjack.Logger{
+		Filename: filepath.Join(ctx.OutputPath, "supply.jsonl"),
+		MaxSize:  200, // megabytes
```
What do you think of adding a flag for this as well @s1na?

s1na replied:
Sure but I think we should change the config and make it similar to native tracers. I.e. we can set the tracer name, aand a JSON string that will be passed to the specific tracer for its configuration. Actually now that we have a concrete use-case I'll go ahead and do that in my branch. Then you can merge and use it.

s1na replied:
Pushed. So now in the CLI can do `--vmtrace.config '{"path": "bla", "maxSize": "200mb"}'` and parse that in the constructor of the supply tracker.

ziogaschr replied:
Thanks @s1na. Much better, here is the change https://github.com/s1na/go-ethereum/pull/21/commits/27bc3911b7fb15768d6e32f674bc319f523132e8

ziogaschr committed:
Merge branch 'extended-tracer' into extended-tracer-chris

ziogaschr committed:
fix live package import issues

ziogaschr commented:
I merged with base repo

s1na commented on eth/tracers/live/supply.go:
```diff
@@ -0,0 +1,247 @@
+	logger      *log.Logger
+
+	// Check if genesis has been processed
+	hasGenesisProcessed bool
```
This shouldn't be necessary anymore (I saw you merged in the latest).

ziogaschr replied:
I will change it tomorrow. Thanks for the note.

ziogaschr replied:
Done with https://github.com/s1na/go-ethereum/pull/21/commits/4dff9da4ca87b46e5621344fa720caabd0442df1

ziogaschr committed:
Revert "add “—vmtrace.output.path” for setting the output path for traces"

This reverts commit fcf0042cda4c7cdf166521b7f4374d974e471009.

ziogaschr committed:
Merge branch 'extended-tracer' into extended-tracer-chris

ziogaschr committed:
parse lumberjack settings from `—vmtrace.config`

ziogaschr committed:
remove unneeded hasGenesisProcessed

s1na committed:
use struct for tracing events

s1na committed:
Merge branch 'extended-tracer' into extended-tracer-struct

karalabe committed:
core/txpool, eth, miner: pre-filter dynamic fees during pending tx retrieval (#29005)

* core/txpool, eth, miner: pre-filter dynamic fees during pending tx retrieval

* miner: fix typo

* core/txpool: handle init-error in blobpool without panicing

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

rjl493456442 committed:
ethstats: prevent panic if head block is not available (#29020)

This pull request fixes a flaw in ethstats which can lead to node crash

A panic could happens when the local blockchain is reorging which causes the original head block not to be  reachable (since number->hash canonical mapping is deleted). In order to prevent the panic, the block nilness is now checked in ethstats.

cuiweixie committed:
core: using math.MaxUint64 instead of 0xffffffffffffffff (#29022)

karalabe committed:
core/txpool, miner: speed up blob pool pending retrievals (#29008)

* core/txpool, miner: speed up blob pool pending retrievals

* miner: fix test merge issue

* eth: same same

* core/txpool/blobpool: speed up blobtx creation in benchmark a bit

* core/txpool/blobpool: fix linter

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

holiman committed:
core/vm, params: ensure order of forks, prevent overflow (#29023)

This PR fixes an overflow which can could happen if inconsistent blockchain rules were configured. Additionally, it tries to prevent such inconsistencies from occurring by making sure that merge cannot be enabled unless previous fork(s) are also enabled.

karalabe committed:
core/txpool, eth, miner: retrieve plain and blob txs separately (#29026)

* core/txpool, eth, miner: retrieve plain and blob txs separately

* core/txpool: fix typo, no farming

* miner: farm all the typos

Co-authored-by: Martin HS <martin@swende.se>

---------

Co-authored-by: Martin HS <martin@swende.se>

buddh0 committed:
core/txpool: fix typo (#29031)

buddh0 committed:
core,eth,internal: fix typo (#29024)

tmelhao committed:
params: add cancun upgrade banner (#29042)

params: add cancun banner

Signed-off-by: tmelhao <tmelhao@gmail.com>
Co-authored-by: tmelhao <tmelhao@gmail.com>

colinlyguo committed:
core/txpool: fix typo (#29036)

* fix typos

* address comments

s1na committed:
internal/ethapi: fix defaults for blob fields (#29037)


Co-authored-by: Martin HS <martin@swende.se>

fjl committed:
params: release go-ethereum v1.13.13 stable

fjl committed:
params: begin v1.13.14 release cycle

s1na committed:
full struct-based tracing infra

s1na committed:
Add canceler interface

ArtificialPB committed:
internal/ethapi: use overriden baseFee for gasPrice (#29051)

eth_call and debug_traceCall allow users to override various block fields, among them base fee. However the overriden base fee was not considered for computing the effective gas price of that message, and instead base fee of the base block was used. This has been fixed in this commit.

colinlyguo committed:
internal/ethapi:fix zero rpc gas cap in eth_createAccessList (#28846)

This PR enhances eth_createAccessList RPC call to support scenarios where the node is launched with an unlimited gas cap (--rpc.gascap 0). The eth_createAccessList RPC call returns failure if user doesn't explicitly set a gas limit.

tmelhao committed:
eth/catalyst: fix wrong error message of payloadV2 after cancun (#29049)

* eth/catalyst: the same error format

Signed-off-by: tmelhao <tmelhao@gmail.com>

* eth/catalyst: wrong error message for payloadV2 post-cancun

Signed-off-by: tmelhao <tmelhao@gmail.com>

* eth/catalyst: parentBeaconBlockRoot -> parentBlockBeaconRoot

Signed-off-by: tmelhao <tmelhao@gmail.com>

* apply commit review

Signed-off-by: tmelhao <tmelhao@gmail.com>

---------

Signed-off-by: tmelhao <tmelhao@gmail.com>
Co-authored-by: tmelhao <tmelhao@gmail.com>

Commit:
core/txpool: reject blob txs with blob fee cap below the minimum (#29081)

* make blobpool reject blob transactions with fee below the minimum

* core/txpool: some minot nitpick polishes and unified error formats

* core/txpool: do less big.Int constructions with the min blob cap

---------

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

0x2d3c committed:
p2p, log, rpc: use errors.New to replace fmt.Errorf with no parameters (#29074)

cuiweixie committed:
eth/catalyst: enable some commented-out testcases   (#29073)

mask-pp committed:
internal/ethapi: pass blob hashes to gas estimation (#29085)

justindhillon committed:
cmd/clef: add spaces in README.md table (#29077)

Add space after links in so they are clickable in vscode.

cuiweixie committed:
eth/catalyst: remove variable in tx conversion loop (#29076)

karalabe committed:
core/txpool/blobpool: reduce default database cap for rollout (#29090)

xcore/txpool/blobpool: reduce default database cap for rollout

jsvisa committed:
core/txpool: no need to log loud rotate if no local txs (#29083)

* core/txpool: no need to run rotate if no local txs

Signed-off-by: jsvisa <delweng@gmail.com>

* Revert "core/txpool: no need to run rotate if no local txs"

This reverts commit 17fab173883168c586d57ca9c05dfcbd9e7831b4.

Signed-off-by: jsvisa <delweng@gmail.com>

* use Debug if todo is empty

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

andreisilviudragnea committed:
eth/tracers: Fix callTracer logs on onlyTopCall == true (#29068)

Commit:
core/txpool: elevate the 'already reserved' error into a constant (#29095)

declare the 'already reserved' error in errors.go

karalabe committed:
params: release Geth v1.13.14

karalabe committed:
params: begin v1.13.15 cycle

s1na committed:
fix memory copy util func

s1na committed:
fix goja scope object wrappers

s1na committed:
fixes, rm t8ntool tracewriter

cuiweixie committed:
core: using math.MaxUint64 instead of 0xffffffffffffffff (#29094)

karalabe committed:
params: being major version bump cycle

rjl493456442 committed:
core, triedb/pathdb: calculate the size for batch pre-allocation (#29106)

* core, triedb/pathdb: calculate the size for batch pre-allocation

* triedb/pathdb: address comment

rjl493456442 committed:
triedb/pathdb: fix panic in recoverable (#29107)

* triedb/pathdb: fix panic in recoverable

* triedb/pathdb: add todo

* triedb/pathdb: rename

* triedb/pathdb: rename

buddh0 committed:
internal/ethapi: pass in accesslist in test (#29089)


Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

s1na committed:
moaar fixes

s1na committed:
renaming GetLogger

s1na committed:
move logger interface to core/tracing

s1na committed:
rename LiveLogger to hooks

s1na committed:
rename GetLogger to Hooks

lightclient committed:
eth/catalyst: return invalid payload attributes instead of invalid parms for bad fcu payload (#29115)

cuishuang committed:
all: fix function names in docs (#29128)

Signed-off-by: cui fliter <imcusg@gmail.com>

weiihann committed:
cmd/geth: parseDumpConfig should not return closed db (#29100)

* cmd: parseDumpConfig should not return closed db

* fix lint

Commit:
all: replace fmt.Errorf() with errors.New() if no param required (#29126)

replace-fmt-errorf

Co-authored-by: yzb@example.cn <yzb@example.cn>

karalabe committed:
cmd/utils, core/rawdb, triedb/pathdb: flip hash to path scheme (#29108)

* cmd/utils, core/rawdb, triedb/pathdb: flip hash to path scheme

* graphql: run tests in hash mode as the chain maker needs it

s1na committed:
eth/catalyst: update simulated beacon for cancun (#28829)

* eth/catalyst: update simulated beacon for cancun

* validate blob hashes

* compute hashes from commitment

* fix beacon root and payload version

* check commitment conversion

* fix random attr

* flip dev to cancun

s1na committed:
rename scopeContext to opContext

s1na committed:
assign const values for gas change reasons

s1na committed:
a type for each hook

s1na committed:
rm old logger file

s1na committed:
rename Capture hooks to On

karalabe committed:
eth: make transaction propagation paths in the network deterministic (#29034)

* eth: make transaction propagation paths in the network deterministic

* eth: avoid potential division by 0

* eth: make tx propagation dependent on local node id too

* eth: fix review comments

undefinedor committed:
all: remove redundant import aliases (#29144)

yasyzb committed:
p2p: remove unused argument 'flags' (#29132)

yasyzb committed:
accounts/abi: delete duplicate error check (#29136)

cuiweixie committed:
core: delete unused ErrMaxInitCodeSizeExceeded (#29062)

buddh0 committed:
all: use EmptyUncleHash, EmptyCodeHash instead of raw value (#29134)

psogv0308 committed:
triedb/pathdb: changed the test code to check for verifying state (#29150)

Co-authored-by: this-is-iron <iron@superblock.co>

fjl committed:
cmd/devp2p: fix commandHasFlag (#29091)

It got broken in some update of the cli library, and thus bootnodes weren't 
being configured automatically for some of the discovery commands.

rjl493456442 committed:
core: use finalized block as the chain freeze indicator (#28683)

* core: use finalized block as the chain freeze indicator

* core/rawdb: use max(finality, head-90k) as chain freezing threshold

* core/rawdb: fix tests

* core/rawdb: fix lint

* core/rawdb: address comments from peter

* core/rawdb: fix typo

andreisilviudragnea committed:
eth/tracers: Fix prestateTracer pre nonce on contract creation (#29099)

The prestateTracer was reporting an inaccurate nonce for the contract being created in
post EIP-158 transactions. Correct nonce is 0, due to the issue nonce was being reported as 1.

reductionista committed:
.mailmap: remove invalid email address (#29163)

vieyang committed:
core/types: reuse signtx (#29152)

* core/types: reuse signtx

* core/types: inline signtx

zhiqiangxu committed:
core: initialize `gasRemaining` with `=` instead of `+=` (#29149)

initialize gasRemaining with = instead of +=

buddh0 committed:
rlp: remove a moot todo (#29154)

rjl493456442 committed:
core/state, trie/triedb/pathdb: remove storage incomplete flag (#28940)

As SELF-DESTRUCT opcode is disabled in the cancun fork(unless the
account is created within the same transaction, nothing to delete
in this case). The account will only be deleted in the following
cases:

- The account is created within the same transaction. In this case
the original storage was empty.

- The account is empty(zero nonce, zero balance, zero code) and
is touched within the transaction. Fortunately this kind of accounts
are not-existent on ethereum-mainnet.

All in all, after cancun, we are pretty sure there is no large contract
deletion and we don't need this mechanism for oom protection.

holiman committed:
accounts/usbwallet: use updated hid (only) library (#28945)

* accounts/usbwallet: use updated hid (only) library

* deps: update karalabe/hid

jsvisa committed:
internal/jsre: format blob fields from hexdecimal to int  (#29166)

* internal/jsre: format receipt.{blobGasPrice,blobGasUsed} to int

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/jsre: format tx.maxFeePerBlobGas to int

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/jsre: format blob* in block

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

undefinedor committed:
rpc: remove deprecated method "Notifier.Closed" (#29162)

cuiweixie committed:
rlp: using unsafe.Slice instead of SliceHeader (#29067)


Co-authored-by: Felix Lange <fjl@twurst.com>

zhiqiangxu committed:
core: remove useless assignments (#29065)

MariusVanDerWijden committed:
node: remove test which doesn't do a lot (#29159)

* node: fix test if directory already exists

* node: remove test

karalabe committed:
consensus, cmd, core, eth: remove support for non-merge mode of operation (#29169)

* eth: drop support for forward sync triggers and head block packets

* consensus, eth: enforce always merged network

* eth: fix tx looper startup and shutdown

* cmd, core: fix some tests

* core: remove notion of future blocks

* core, eth: drop unused methods and types

Commit:
go.mod: bump pebble db to official release (#29038)

bump pebble

tomdever committed:
internal/ethapi: delete needless error check (#29127)

holiman committed:
accounts/usbwallet: revert #28945 (#29175)

AyumuKasuga committed:
rpc: add more test cases for arg types (#29006)

holiman committed:
accounts/usbwallet: update hid library (#29176)

undefinedor committed:
accounts: remove deprecated function NewPlaintextKeyStore (#29171)

jsvisa committed:
eth/tracers: prestate tracer add blob fee (#29168)

* eth/tracers: prestate balance add blob fee

Signed-off-by: jsvisa <delweng@gmail.com>

* eth/tracers: prestate test support blob tx

Signed-off-by: jsvisa <delweng@gmail.com>

* eth/tracers: add prestate blob tx test

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

MariusVanDerWijden committed:
miner: refactor the miner, make the pending block on demand (#28623)

* miner: untangle miner

* miner: use common.hash instead of *types.header

* cmd/geth: deprecate --mine

* eth: get rid of most miner api

* console: get rid of coinbase in welcome message

* miner/stress: get rid of the miner stress test

* eth: get rid of miner.setEtherbase

* ethstats: remove miner and hashrate flags

* ethstats: remove miner and hashrate flags

* cmd: rename pendingBlockProducer to miner.pending.feeRecipient flag

* miner: use pendingFeeRecipient instead of etherbase

* miner: add mutex to protect the pending block

* miner: add mutex to protect the pending block

* eth: get rid of etherbase mentions

* miner: no need to lock the coinbase

* eth, miner: fix linter

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Péter Szilágyi <peterke@gmail.com>

s1na committed:
rm OnStart and onEnd

s1na committed:
rename tracer methods to On pattern

s1na committed:
rm embedding of noop tracer

s1na committed:
rm OnKeccakPreimage

zsfelfoldi committed:
cmd/blsync, beacon/light: beacon chain light client (#28822)

Here we add a beacon chain light client for use by geth.

Geth can now be configured to run against a beacon chain API endpoint,
without pointing a CL to it. To set this up, use the `--beacon.api` flag. Information
provided by the beacon chain is verified, i.e. geth does not blindly trust the beacon
API endpoint in this mode. The root of trust are the beacon chain 'sync committees'.

The configured beacon API endpoint must provide light client data. At this time, only
Lodestar and Nimbus provide the necessary APIs.

There is also a standalone tool, cmd/blsync, which uses the beacon chain light client
to drive any EL implementation via its engine API.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

s1na committed:
fix eth config

s1na committed:
replace opcode with byte

s1na committed:
add fn refundGas

s1na committed:
rm canceler

cuinix committed:
accounts: remove redundant string conversion (#29184)

hyhnet committed:
all: fix typos in comments (#29186)

sebastianst committed:
log: add Handler getter to Logger interface (#28793)

log: Add Handler getter to Logger interface

colinlyguo committed:
ethclient/gethclient: add blob transaction fields in toCallArg (#29198)

tmelhao committed:
console: fix the wrong error msg of datadir testcase (#29183)

keroro520 committed:
p2p/simulations/adapters: fix error messages in TestTCPPipeBidirections (#29207)

karalabe committed:
cmd, core, metrics: always report expensive metrics (#29191)

* cmd, core, metrics: always report expensive metrics

* core, metrics: report block processing metrics as resetting timer

* metrics: update reporter tests

PlasmaPower committed:
core/txpool/blobpool: return ErrAlreadyKnown for duplicate txs (#29210)

Signed-off-by: Lee Bousfield <ljbousfield@gmail.com>

s1na committed:
core: fix deprecation comment for GenesisAccount (#29218)

core: fix deprecation comment

testwill committed:
beacon/light/request: fix typos (#29216)

s1na commented on eth/tracers/live/supply.go:
```diff
@@ -0,0 +1,260 @@
+	s.delta = newSupplyInfo()
+}
+
+func (s *Supply) OnBlockStart(b *types.Block, td *big.Int, finalized, safe *types.Header, skip bool) {
```
Writing so we don't forget. We need to add blob gas burnt here as well.

kumakichi committed:
crypto/bn256/cloudflare: fix noescape-directive (#29222)

songzhibin97 committed:
cmd: use package filepath over path for file system operations (#29227)

Package filepath implements utility routines for manipulating filename paths in a way compatible with the target operating system-defined file paths.

Package path implements utility routines for manipulating slash-separated paths.

The path package should only be used for paths separated by forward slashes, such as the paths in URLs

wzshiming committed:
internal/build, rpc: add missing HTTP response body Close() calls (#29223)


Co-authored-by: Felix Lange <fjl@twurst.com>

AaronChen0 committed:
common/math: copy result in Exp (#29233)

common/math: does not change base parameter

MariusVanDerWijden committed:
miner: add additional log (#29193)

Adds a debug level log if the payload building failed for whatever reason

s1na committed:
resolve merge conflict

fjl committed:
p2p: fix race in dialScheduler (#29235)


Co-authored-by: Stefan <stefan@starflinger.eu>

s1na committed:
eth/tracers: fix concurrency issue for JS-tracing a block (#29238)

This change fixes a concurrency-issue where JS-tracers were accessing the block-ctx GetHash function in a in parallel, which is not safe.

holiman committed:
crypto/kz4844: pass blobs by ref (#29050)

This change makes use of the following underlying changes to the kzg-libraries in order to avoid passing large things on the stack:

- c-kzg: https://github.com/ethereum/c-kzg-4844/pull/393 and
- go-kzg: https://github.com/crate-crypto/go-kzg-4844/pull/63

justindhillon committed:
accounts/usbwallet, common/bitutil: fix broken links in docs (#29078)

fixes some links in documentation

rjl493456442 committed:
core: improve chain rewinding mechanism (#29196)

* core: improve chain rewinding mechanism

* core: address comment

* core: periodically print progress log

* core: address comments

* core: fix comment

* core: fix rewinding in path

* core: fix beyondRoot condition

* core: polish code

* core: polish code

* core: extend code comment

* core: stop rewinding if chain is gapped or genesis is reached

* core: fix broken tests

s1na committed:
fix lint issue

s1na committed:
fix merge conflict

s1na committed:
fix merge conflict

s1na committed:
update supply tracer to new interface

s1na committed:
rm captureEnd

s1na committed:
add blob burnt fee

s1na committed:
minor

s1na committed:
rm extra param

holiman committed:
tests, appveyor: only execute one in four permutations on CI (#29220)

tests, appveyor: only execute one in four permutations when flag -short is used

Also enable -short flag on all appveyor builds (also ubuntu)

songzhibin97 committed:
go.mod: update golang.org/x/crypto from v0.17.0 to v0.21.0 (#29228)

tmelhao committed:
eth/catalyst: remove error return in delayPayloadImport (#29043)


Co-authored-by: tmelhao <tmelhao@gmail.com>

weiihann committed:
cmd/devp2p: fix decoding of raw RLP ENR attributes (#29257)

maoueh committed:
Call `OnBlockchainInit` before `OnGenesisBlock` (#22)

s1na committed:
fix empty tracer config

dyxushuai committed:
cmd/emv/internal/t8ntool: fix shadowing of `excessBlobGas` (#29263)

fix(t8n): unexpected `excessBlobGas` shadowed

s1na committed:
simplify eip1559 test

tmelhao committed:
params: use the same variable name as EIP-4788 (#29195)

In https://eips.ethereum.org/EIPS/eip-4788 the name `BEACON_ROOTS_ADDRESS` is used. This change makes geth use the same variable name to avoid confusion.

shivhg committed:
eth/downloader, graphql: fix typos (#29243)

holiman committed:
eth/catalyst: prettier output on bad new payloads (#29259)

When we receive a bad NewPayload, we currently emit a lot of data to the logging facilities. This PR makes it so we print less data.

kumakichi committed:
accounts/abi/bind: remove unused err set and check (#29269)

accounts/abi: remove unused err set and check

s1na committed:
refactors

holiman committed:
go.mod: update protobuf (#29270)

holiman committed:
cmd/evm: fix tracing

This commit fixes several missing interface-checks, where the code did not properly
verify a Hook before calling it.
It also fixes an error where the tracer was not set in the vmconfig during t8n initialization.

It also adds a testcase which verifies the tracing-output during t8n executio. The
testcase currently only checks the jsonlogger, but can be extended.

This change also removes a spurious 'null' output in the tracing, when
tracing result is null.

holiman committed:
cmd/evm: add testcase for js-tracer

holiman committed:
core/vm: documentation + minor refactoring

holiman committed:
eth/tracers, cmd/evm: unexport JSONlogger

holiman committed:
all: update to go version 1.22.1 (#28946)

Since Go 1.22 has deprecated certain elliptic curve operations, this PR removes 
references to the affected functions and replaces them with a custom implementation
in package crypto. This causes backwards-incompatible changes in some places.

---------

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>
Co-authored-by: Felix Lange <fjl@twurst.com>

rjl493456442 committed:
triedb/pathdb: improve tests (#29278)

AaronChen0 committed:
log: replace tmp with bytes.Buffer.AvailableBuffer (#29287)

AtomicInnovation321 committed:
beacon/light/sync: fix typo in comment (#29256)

htiennv committed:
accounts/abi/bind: check invalid chainID first (#29275)

buddh0 committed:
miner: modify header before checking time-based fields (#29242)

The Prepare-method of consensus engine might modify the time-field in a header, so it should be called prior to checks that rely on it

s1na committed:
refactor args to msg and tx

zgfzgf committed:
core/asm: minor code-clarification (#29293)

miles-six committed:
all: fix typos (#29288)

georgehao committed:
metrics: fix docstrings (#29279)

holiman committed:
cmd/evm: make staterunner always output stateroot to stderr (#29290)

This changes makes it so that when `evm statetest` executes, regardless of whether `--json` is specified or not, the stateroot is printed on `stderr` as a `jsonl` line. This enables speedier execution of testcases in goevmlab, in cases where full execution op-by-op is not required.

holiman committed:
cmd/evm: fix flag-mismatch from #29290 (#29298)

imalasong committed:
Makefile: update PHONY directive (#29296)

rjl493456442 committed:
core: skip the check the statefulness of head block in repair (#29245)

AaronChen0 committed:
core/txpool/blobpool: calculate log1.125 faster (#29300)

MariusVanDerWijden committed:
core: refactor consensus interface (#29283)

This PR modifies the consensus interface to wrap the body fields.

holiman committed:
eth/protocols/snap, internal/testlog: fix dataraces (#29301)

s1na committed:
fix merge conflict

s1na committed:
Merge branch 'extended-tracer-struct' into extended-tracer-chris

s1na commented:
Closing in favor of https://github.com/ethereum/go-ethereum/pull/29347
